src/antelope/stores/evm.ts

```diff
@@ -63,161 +55,131 @@ export const evmEvents = {
+                    window.removeEventListener('focus', checkNetworkHandler);
+                    if (useAccountStore().loggedAccount) {
+                        window.addEventListener('focus', checkNetworkHandler);
                     }
-                    return accessGranted[0];
-                }
-            } catch (error) {
-                if ((error as unknown as ExceptionError).code === 4001) {
-                    throw new AntelopeError('antelope.evm.error_connect_rejected');
-                } else {
-                    console.error('Error:', error);
-                    throw new AntelopeError('antelope.evm.error_login');
-                }
-            } finally {
-                useFeedbackStore().unsetLoading('evm.login');
-                const provider = await this.ensureProvider();
-                const checkProvider = new ethers.providers.Web3Provider(provider);
-                const signer = await checkProvider.getSigner();
-                this.setExternalSigner(signer);
+                });
 
-            }
-        },
-        async sendSystemToken (to: string, value: BigNumber): Promise<EvmTransactionResponse> {
-            this.trace('sendSystemToken', to, value);
-
-            // Send the transaction
-            if (this.signer) {
-                return this.signer.sendTransaction({
-                    to,
-                    value,
-                }).then(
-                    (transaction: ethers.providers.TransactionResponse) => transaction,
-                ).catch((error) => {
-                    if ('ACTION_REJECTED' === ((error as {code:string}).code)) {
-                        throw new AntelopeError('antelope.evm.error_transaction_canceled');
+                provider.on('accountsChanged', async (value) => {
+                    const accounts = value as string[];
+                    const network = useChainStore().currentChain.settings.getNetwork();
+                    evm.trace('provider.accountsChanged', ...accounts);
+
+                    // accounts.length > 0 means has just logged in or is switching account in wallet
+                    if (accounts.length > 0) {
+                        if (useAccountStore().loggedAccount) {
+                            // if the user is already authenticated we try to re login the account using the same authenticator
+                            const authenticator = useAccountStore().loggedAccount.authenticator as EVMAuthenticator;
+                            if (!authenticator) {
+                                console.error('Inconsistency: logged account authwenticator is null', authenticator);
+                            } else {
+                                useAccountStore().loginEVM({ authenticator,  network });
+                            }
+                        } else {
+                            // if not, we do nothing because this is part of the happy ligin path
```

Comment:
```suggestion
                            // if not, we do nothing because this is part of the happy login path
```

Reply:
thanks! I'm removing the else part enyway